lib/modules/dosomething/dosomething_api/includes/Transformer.php

```diff
@@ -83,39 +99,19 @@ protected function getPathParameters($parameters, $endpoint) {
 
 
   /**
-   * Calculate total number of Reportback items for the specified campaigns by
-   * each status requested.
-   *
-   * @param array $parameters Query parameters.
+   * @param $ids
    *
-   * @return int Total number of reportback items.
-   */
-  protected function countItems($parameters) {
-    $total = 0;
-
-    foreach((array) $parameters['nid'] as $id) {
-      foreach((array) $parameters['status'] as $status) {
-        $total += (int) dosomething_reportback_get_reportback_total_by_status($id, $status);
-      }
-    }
-
-    return $total;
-  }
-
-
-  /**
-   * @param string $data Single or multiple comma separated data items.
-   *
-   * @return string or array
+   * @return array
    */
-  protected function formatData($data) {
-    $array = explode(',', $data);
+  protected function getReportbackItems($ids) {
+    $filters = array(
+      'fid' => $this->formatData($ids),
+    );
 
-    if (count($array) > 1) {
-      return $array;
-    }
+    // Obtaining all Reportback items.
+    $query = dosomething_reportback_get_reportback_files_query_result($filters, 'all');
```

Comment:
hahahahah. Yeah, I hear you though and agree. We could also potentially keep it in the ReportbackTransformer and this Transformer somehow instantiates the ReportbackTransformer class to get access to those reportback functions, but not sure if that starts to get too hairy. Definitely open to a future refactor once we get some of these other transformers in play and see how they interact.